baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/AbstractDeprecatedApiCheck.java

```diff
@@ -88,9 +110,39 @@ private Description checkTree(Tree tree, VisitorState state) {
         return buildDescription(tree).setMessage(description).build();
     }
 
-    private Optional<Name> getCurrentClass(VisitorState state) {
-        return Optional.ofNullable(ASTHelpers.findEnclosingNode(state.getPath(), ClassTree.class))
-                .map(ASTHelpers::getSymbol)
-                .map(Symbol::getQualifiedName);
+    private Optional<ClassSymbol> getOwningClass(Optional<Symbol> symbol) {
+        if (symbol.isEmpty()) {
+            return Optional.empty();
+        }
```

Comment:
I would just take a `Symbol` here and use `.flatMap` in the place that has an `Optional<Symbol>`. Just feels weird to take an `Optional` when empty is not a useful parameter to this method.